presto-iceberg/pom.xml

```diff
@@ -41,6 +41,11 @@
             <artifactId>presto-hive</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>io.prestosql</groupId>
+            <artifactId>presto-main</artifactId>
```

Comment:
Would `InMemoryRecordSet` be less efficient than `InternalTable`? because of the conversion in https://github.com/prestosql/presto/blob/51131ff1da9902790d6e472ffbc6025b20c76bb9/presto-spi/src/main/java/io/prestosql/spi/connector/RecordPageSource.java#L95-L121

Is it reasonable to move `InternalTable` to `presto-spi`?

Reply:
The data is expected to be tiny, so it's doesn't really matter. `InternalTable` does the same conversion on write using `writeNativeValue`. Note that `RecordPageSource` is not used in `ScanFilterAndProjectOperator` since it detects that and generates code that's specialized for each column.

Something like `PageListBuilder` in Raptor is better, since you're building pages directly with the correct type. I think Raptor v2 (not yet merged) has an improved version of this.